File: configor/src/main/java/com/bruceewu/configor/holder/base/FlowHolder.java
package com.bruceewu.configor.holder.base;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.bruceewu.configor.IConfigor;
import com.bruceewu.configor.R;
import com.bruceewu.configor.entity.CusOnClickListener;
import com.bruceewu.configor.entity.DisplayItem;
import com.bruceewu.configor.entity.IDivider;

public class FlowHolder extends CusBaseHolder implements IDivider {

    public FlowHolder(View rootView) {
        super(rootView);
    }

    @Override
    public int layoutID() {
        return R.layout.holder_flow;
    }

    @Override
    public void renderUI(DisplayItem item, CusOnClickListener listener) {
        ((FlowLayout) mHelper.getView(R.id.flow_layout)).set(15, 10, item.children(), listener);
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.left = IConfigor.configor().dip2px(15);
    }
}

File: configor/src/main/java/com/bruceewu/configor/holder/base/FlowHolder.kt
package com.bruceewu.configor.holder.base

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bruceewu.configor.IConfigor
import com.bruceewu.configor.R
import com.bruceewu.configor.entity.CusOnClickListener
import com.bruceewu.configor.entity.DisplayItem
import com.bruceewu.configor.entity.IDivider

class FlowHolder(itemView: View) : CusBaseHolder(itemView), IDivider {
    private val space = IConfigor.configor.dip2px(15)

    override fun layoutID() = R.layout.holder_flow

    override fun renderUI(item: DisplayItem, listener: CusOnClickListener) {
        mHelper.getView<FlowLayout>(R.id.flow_layout).set(15, 10, item.children(), listener)
    }

    override fun getItemOffsets(outRect: Rect, view: View?, parent: RecyclerView?, state: RecyclerView.State?) {
        outRect.left = space
    }
}